e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java

```diff
@@ -0,0 +1,491 @@
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.hamcrest.Matchers.matchesPattern;
+
+// Unit tests here can be run from the IDE and will use LOCAL as the default, they can also be run from the TestLauncher
+// class to specify a custom environment
+@Testcontainers
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ExtendWith(TestRunnerParameterResolver.class)
+public class TestRunner {
+
+    private static HttpClient httpClient;
+
+    private static String AB2D_API_URL;
+
+    private static String AB2D_ADMIN_URL;
+
+    private static final String PATIENT_EXPORT_PATH = "Patient/$export";
+
+    private static String jwtStr = null;
+
+    private static final long DEFAULT_TIMEOUT = 30;
+
+    private static final int DELAY = 30;
+
+    private static final int JOB_TIMEOUT = 300;
+
+    private Environment environment;
+
+    public void runTests() throws InterruptedException, JSONException, IOException {
+        runSystemWideExport();
+        runContractNumberExport();
+        testDelete();
+    }
+
+    public TestRunner(Environment environment) throws IOException, InterruptedException, JSONException {
+        this.environment = environment;
+        init();
+    }
+
+    public void init() throws IOException, InterruptedException, JSONException {
+        if(environment.isUsesDockerCompose()) {
+            DockerComposeContainer container = new DockerComposeContainer(
+                    new File("../docker-compose.yml"))
+                    //.withScaledService("api", 2) // failing now since it's not changing ports
+                    .withScaledService("worker", 2)
+                    .withExposedService("db", 5432)
+                    .withExposedService("api", 8080, new HostPortWaitStrategy()
+                            .withStartupTimeout(Duration.of(120, SECONDS)));
+            container.start();
+        }
+
+        Yaml yaml = new Yaml();
+        InputStream inputStream = getClass().getResourceAsStream("/" + environment.getConfigName());
+        Map<String, String> yamlMap = yaml.load(inputStream);
+        String oktaUrl = yamlMap.get("okta-url");
+
+        AB2D_API_URL = yamlMap.get("ab2d-api-url");
+        AB2D_ADMIN_URL = yamlMap.get("ab2d-admin-url");
+
+        var jwtRequestParms = new HashMap<>() {{
+            put("grant_type", "client_credentials");
+            put("scope", "clientCreds");
+        }};
+
+        String oktaClientId = System.getenv(environment.name() + "_OKTA_CLIENT_ID");
+        String oktaPassword = System.getenv(environment.name() + "_OKTA_CLIENT_PASSWORD");
```

Comment:
The names of the environment variables should _not_ change across tiers -- it is the values that change. Making variable _names_ tier-dependent will make Jenkins setup more difficult.